src/transforms/filter.js

```diff
@@ -138,10 +138,16 @@ exports.supplyDefaults = function(transformIn) {
     var enabled = coerce('enabled');
 
     if(enabled) {
+        var target = coerce('target');
+
+        if(Lib.isArrayOrTypedArray(target) && target.length === 0) {
```

Comment:
@etpinard Thanks for the PR.
Cool! 
What about handling a case like this: `[ [ ] ]`?  
See [codepen](https://codepen.io/MojtabaSamimi/pen/PgKYpr).

Reply:
Good call. Thanks for the review!

Reply:
@archmoj this turned out to be a bit more of a project than anticipated. 

See https://github.com/plotly/plotly.js/pull/3766/commits/3a5a4c2b2761dd232838b6c59109650a308a3077 - which makes things work for all trace types except `scattercarpet` and in some world-calendar scenario. I'll try to get those two cases fixed before `1.48.0`, but in the meantime here's "a good chuck" of the solution.

Reply:
> which makes things work for all trace types except `scattercarpet` and in some world-calendar scenario.

Now fixed by https://github.com/plotly/plotly.js/pull/3766/commits/b6432380de4b3474b0bec9b2f02d2bd49e416f15 and https://github.com/plotly/plotly.js/pull/3766/commits/40c8abc10cb029d1300c6b0c696acd71299bb3c9

Making this PR ready for a second :eye: